doc: dts: api: update the inter-node dependencies documentation (#64463)
zephyrproject-rtos/zephyr: doc/build/dts/api/api.rst

- a node directly depends on any nodes its properties refer to by phandle
- a node directly depends on its ``interrupt-parent`` if it has an
  ``interrupts`` property

> - a node with child nodes also inherits all the dependencies of its child nodes

Reviewer: This "parent inherits from children" wording is quite confusing but I am not sure I have a good alternative. Maybe the following is slightly better tho?

```suggestion
- a parent node inherits all dependencies from its child nodes.
```

Or

```suggestion
- child node dependencies are also counted as dependencies for their parent node.
```

Author: "a parent node inherits all dependencies from its child node" sounds better, let's try and go with that. Also wondering if this should specific "from its child nodes that don't have their own binding", since that's how it actually works.
